copy/copy.go

```diff
@@ -84,7 +84,7 @@ type imageCopier struct {
 	manifestUpdates    *types.ManifestUpdateOptions
 	src                types.Image
 	diffIDsAreNeeded   bool
-	canModifyManifest  bool
+	canModifyManifest  string
```

Comment:
(Others: WDYT?)

Hum, I didn’t anticipate just using a string without a bool for this. But it works pretty well? Except that it’s non-obvious without understanding the full picture.

Either split this into `canModifyManifest bool` `cannotModifyManifestReason string`, or at least rename and document the strings to something like
```go
cannotModifyManifestReason string // Either "" if modification is allowed, or a reason to include in error messages
```

(Similarly for the `canModifyManifestList` variable.)

Reply:
I feel like it's fairly obvious by reading the code, but am happy to change it up to suit. I couldn't think of a _neat_ way to do it, which is one of the reasons I didn't do it in the first commit :)

Which would you prefer, two variables, or one with a better name and comment?

Reply:
Basically I’m worried `if canModifyManifest != ""` is non-obvious (boolean naming + string type, “true” indicated by empty string) for busy people seeing a few lines in isolation, e.g. in some search results.

On balance, I’d probably prefer just the `cannot…Reason` name (or some better name… `fixedManifestReason`??? I’m bad at naming) + comment, because that way the two can’t get out of sync. I didn’t originally realize the …reason variable would probably be named the same in both cases.

(But this is all a somewhat weak preference, if others feel differently, please speak up.)

Reply:
Yep, and I share your concern, it could be more obvious.

Have gone with one variable that is more appropriately named and better commented.